app/models/mixins/has_infra_manager_mixin.rb

```diff
@@ -17,13 +17,6 @@ def virtualization_endpoint_destroyed(role)
   private
 
   def ensure_infra_manager
-    if infra_manager.nil?
-      build_infra_manager(:parent_manager  => self,
-                          :name            => "#{name} Virtualization Manager",
-                          :zone_id         => zone_id,
-                          :provider_region => provider_region)
-    end
-
-    infra_manager
+    infra_manager ||= propagate_child_attributes(build_infra_manager(:name => "#{name} Virtualization Manager"))
```

Comment:
I think we want to update the child attributes even after it is build to keep things in sync so:
```
infra_manager ||= build_infra_manager
propegate_child_attributes(infra_manager)
```